proxy/proxy.go

```diff
@@ -872,8 +872,7 @@ func (p *Proxy) makeUpgradeRequest(ctx *context, req *http.Request) error {
 func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Context) (*http.Response, *proxyError) {
 	req, endpoint, err := mapRequest(ctx, requestContext, p.flags.HopHeadersRemoval())
 	if err != nil {
-		p.log.Errorf("could not map backend request, caused by: %v", err)
-		return nil, &proxyError{err: err}
+		return nil, &proxyError{err: fmt.Errorf("could not map backend request, caused by: %w", err)}
```

Comment:
This kind of wrapping means we can not type assert on `proxyError.err` field anymore. Maybe its fine, I could not find any usage accept in `NetError()` which is not used so lets remove `NetError()` to reduce confusion.

Reply:
Maybe I should just use error.Is and not use type casting :)

Reply:
But I think we don't use it so lets remove, the type is private anyway

Reply:
https://go.dev/play/p/KA1Y0T8u3Dq

Reply:
I think we can omit `, caused by` in the message for clarity

Reply:
done